internal/command/deploy/machines.go

```diff
@@ -495,6 +503,38 @@ func (md *machineDeployment) configureLaunchInputForReleaseCommand(launchInput *
 	if _, present := launchInput.Config.Env["RELEASE_COMMAND"]; !present {
 		launchInput.Config.Env["RELEASE_COMMAND"] = "1"
 	}
+	if launchInput.Config.Guest == nil {
+		launchInput.Config.Guest = &api.MachineGuest{}
+	}
+	desiredMem := 256
+	if !md.machineSet.IsEmpty() {
+		leastRamInGroup := func(group string) int {
+			leastRam := 0
+			for _, lm := range md.machineSet.GetMachines() {
+				mach := lm.Machine()
+				if mach.Config != nil && mach.Config.Env != nil && mach.ProcessGroup() == group && mach.Config.Guest != nil {
+					if mach.Config.Guest.MemoryMB < leastRam || leastRam == 0 {
+						leastRam = mach.Config.Guest.MemoryMB
+					}
+				}
+			}
+			return leastRam
+		}
+
+		if leastRamApp := leastRamInGroup("app"); leastRamApp != 0 {
+			desiredMem = leastRamApp
+		} else {
+			// No "main" app, sort lexicographically and just pick the first group
+			pgNames := lo.Keys(md.processConfigs)
+			if len(pgNames) > 0 {
+				slices.Sort(pgNames)
+				if leastRamApp = leastRamInGroup(pgNames[0]); leastRamApp != 0 {
+					desiredMem = leastRamApp
+				}
+			}
+		}
```

Comment:
So basically, this looks at each machine in "app" and takes the least amount of ram.
If there's no "app" process group, it sorts the list of process groups and runs the same algorithm through the first group.

Fallback is 256 mb